src/formatDate.js

```diff
@@ -8,7 +8,66 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYear = yearFormatAndPlace(fromFormat);
+  const newYear = yearFormatAndPlace(toFormat);
+
+  const formatedDate = [];
+  let year = '';
+
+  const dateParts = date.split(fromFormat[3]);
+  const newDateSeparator = toFormat[3];
+
+  const thresholdYear = 30;
+
+  if (oldYear.format !== newYear.format) {
+    if (oldYear.format.length < newYear.format.length) {
+      year =
+        dateParts[oldYear.place] < thresholdYear
+          ? `20${dateParts[oldYear.place]}`
+          : `19${dateParts[oldYear.place]}`;
+    } else {
+      year = dateParts[oldYear.place].slice(2);
+    }
+  } else {
+    year = dateParts[oldYear.place];
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i]) {
+      case 'YYYY':
+      case 'YY':
+        formatedDate[i] = year;
+        break;
+      case 'MM':
+        formatedDate[i] = dateParts[fromFormat.indexOf('MM')];
```

Comment:
The use of `fromFormat.indexOf('MM')` assumes that 'MM' will be present in the `fromFormat` array. If 'MM' is not a part of `fromFormat`, this will return -1, and `dateParts[-1]` will be undefined, which will cause an error in the output.